packages/@aws-cdk/aws-lambda/lib/function-url.ts

```diff
@@ -0,0 +1,217 @@
+   */
+  readonly allowedHeaders?: string[];
+
+  /**
+   * An HTTP method that you allow the origin to execute.
+   *
+   * @default - No methods allowed.
+   */
+  readonly allowedMethods?: HttpMethods[];
+
+  /**
+   * One or more origins you want customers to be able to access the bucket from.
+   *
+   * @default - No origins allowed.
+   */
+  readonly allowedOrigins?: string[];
+
+  /**
+   * One or more headers in the response that you want customers to be able to access from their applications.
+   *
+   * @default - No headers exposed.
+   */
+  readonly exposedHeaders?: string[];
+
+  /**
+   * The time in seconds that your browser is to cache the preflight response for the specified resource.
+   *
+   * @default - No caching.
+   */
+  readonly maxAge?: Duration;
+}
+
+/**
+ * A Lambda function Url
+ */
+export interface IFunctionUrl extends IResource {
+  /**
+   * The url of the Lambda function.
+   *
+   * @attribute FunctionUrl
+   */
+  readonly url: string;
+
+  /**
+   * The ARN of the function this URL refers to
+   *
+   * @attribute FunctionArn
+   */
+  readonly functionArn: string;
+}
+
+/**
+ * Options to add a url to a Lambda function
+ */
+export interface FunctionUrlOptions {
+  /**
+   * The type of authentication that your function URL uses.
+   */
+  readonly authType: FunctionUrlAuthType;
+
+  /**
+   * The cross-origin resource sharing (CORS) settings for your function URL.
+   *
+   * @default - No CORS configuration.
+   */
+  readonly cors?: FunctionUrlCorsOptions;
+}
+
+/**
+ * Properties for a FunctionUrl
+ */
+export interface FunctionUrlProps extends FunctionUrlOptions {
+  /**
+   * The function to which this url refers.
+   * It can also be an `Alias` but not a `Version`.
+   */
+  readonly function: IFunction;
+}
+
+/**
+ * Defines a Lambda function url
+ *
+ * @resource AWS::Lambda::Url
+ */
+export class FunctionUrl extends Resource implements IFunctionUrl {
+  /**
+   * The url of the Lambda function.
+   */
+  public readonly url: string;
+
+  /**
+   * The ARN of the function this URL refers to
+   */
+  public readonly functionArn: string;
+
+  /**
+   * The function to which this url refers.
+   */
+  public readonly function: IFunction;
+
+  constructor(scope: Construct, id: string, props: FunctionUrlProps) {
+    super(scope, id);
+
+    if (this.instanceOfVersion(props.function)) {
+      throw new Error('FunctionUrl cannot be used with a Version');
+    }
+
+    let qualifier: string | undefined = undefined;
+    if (this.instanceOfAlias(props.function)) {
+      qualifier = props.function.aliasName;
+    }
+
+    const resource: CfnUrl = new CfnUrl(this, 'Resource', {
+      authType: props.authType,
+      targetFunctionArn: props.function.functionArn,
+      qualifier,
+      cors: props.cors ? this.renderCors(props.cors) : undefined,
+    });
+
+    this.url = resource.attrFunctionUrl;
+    this.functionArn = resource.attrFunctionArn;
+    this.function = props.function;
```

Comment:
Why do you need this public `function` property?

Reply:
No particular reason but initially thought that `Function` must be reachable from the `FunctionUrl`. I have removed it for now since that might not be required in most cases.